### ISSUE TITLE
AttributeError: 'int' object has no attribute 'upper' for az webapp up command

### ISSUE THREAD

**Issue by shimesaba-type0** [Azure/azure-cli#11221]
## Describe the bug

**Command Name**
`az webapp up`

**Errors:**
```
'int' object has no attribute 'upper'
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 603, in execute
    raise ex
cli/core/commands/__init__.py, ln 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 652, in _run_job
    cmd_copy.exception_handler(ex)
...
cli/command_modules/appservice/custom.py, ln 2993, in webapp_up
    create_app_service_plan(cmd, rg_name, plan, _is_linux, False, sku, 1 if _is_linux else None, location)
cli/command_modules/appservice/custom.py, ln 1390, in create_app_service_plan
    sku = _normalize_sku(sku)
cli/command_modules/appservice/utils.py, ln 20, in _normalize_sku
    sku = sku.upper()
AttributeError: 'int' object has no attribute 'upper'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az webapp up --name {} --resource-group {} --plan {} --sku {} --location {}`

## Environment Summary
```
Linux-4.15.0-1061-azure-x86_64-with-debian-stretch-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.76

```

**Comment by viswanath27** [comment]
Facing same issue when I am deploying the flask app in Azure webapp. 

`
'int' object has no attribute 'upper'
Traceback (most recent call last):
File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 603, in execute
    raise ex
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 652, in _run_job
    cmd_copy.exception_handler(ex)
File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 55, in _polish_bad_errors
    raise ex
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 631, in _run_job
    result = cmd_copy(params)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 306, in __call__
    return self.handler(*args, **kwargs)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 485, in default_command_handler
    return op(**command_args)
File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2993, in webapp_up
    create_app_service_plan(cmd, rg_name, plan, _is_linux, False, sku, 1 if _is_linux else None, location)
File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 1390, in create_app_service_plan
    sku = _normalize_sku(sku)
File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/utils.py", line 20, in _normalize_sku
    sku = sku.upper()
AttributeError: 'int' object has no attribute 'upper'
`

command used is 
`az webapp up --sku F1 -n <webappname> -l "Central US" `

[link](https://docs.microsoft.com/en-us/azure/app-service/containers/quickstart-python?tabs=bash) is used for deploying.

**Comment by viswanath27** [comment]
How ever when I run --dryrun command option it runs fine

`az webapp up -n <webappname> --dryrun`

**Comment by panchagnula** [comment]
this is fixed by https://github.com/Azure/azure-cli/pull/11019

**Comment by viswanath27** [comment]
Changing the azure-cli version to 2.0.75, is the solution for this[could install in my pc]. I am not able continue after doing so. After installing `No module named azure` error is showing up. Please direct to the fix (which I can use either in local or in cloud shell) or is it something that it will be available only after Nov26th (2019). Need to complete something.

**Comment by tthaliath** [comment]
Hi I just installed azure cli today on mac. Apparently the fix #11019 doesn't seem to be working. 

webapp xxx-hw doesn't exist
The command failed with an unexpected error. Here is the traceback:

'int' object has no attribute 'upper'
Traceback (most recent call last):
  File "/Users/<username>/lib/azure-cli/lib/python2.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/Users/<username>/lib/azure-cli/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 603, in execute
    raise ex
AttributeError: 'int' object has no attribute 'upper'

Result from dryrun (works fine)

/Users/<username>/bin/az webapp up -n xxx-hw --dryrun
webapp xxx-hw doesn't exist
Web app will be created with the below configuration,re-run command without the --dryrun flag to create & deploy a new app
{
  "appserviceplan": "<userid>_asp_Linux_centralus_0",
  "location": "centralus",
  "name": "xxx-hw",
  "os": "Linux",
  "resourcegroup": "<userid>_rg_Linux_centralus",
  "runtime_version": "python|3.7",
  "sku": "PREMIUMV2",
  "src_path": "//Users//<username>//azure//python-docs-hello-world",
  "version_detected": "-"
}


Any suggesions ? Thanks

**Comment by tthaliath** [comment]
please treat it as urgent as i stuck in a new project.

az webapp up -n username-hello-world -l centralus
webapp username-hello-world doesn't exist
The command failed with an unexpected error. Here is the traceback:

'int' object has no attribute 'upper'
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 603, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 652, in _run_job
    cmd_copy.exception_handler(ex)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/commands.py", line 55, in _polish_bad_errors
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 631, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 306, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/core/__init__.py", line 485, in default_command_handler
    return op(**command_args)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 2993, in webapp_up
    create_app_service_plan(cmd, rg_name, plan, _is_linux, False, sku, 1 if _is_linux else None, location)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 1390, in create_app_service_plan
    sku = _normalize_sku(sku)
  File "/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/utils.py", line 20, in _normalize_sku
    sku = sku.upper()
AttributeError: 'int' object has no attribute 'upper'

**Comment by panchagnula** [comment]
@tthaliath from the stack looks like you are still using the version /azure-cli/2.0.76 that doesn't have the fix. Please downgrade to the previous version 2.0.75

pip install azure-cli==2.0.75

**Comment by tthaliath** [comment]
Thanks sisira for the response.  I have a workaround to resolve this issue without downgrading AZURE client. Since I am using a free tier account of Azure, the SKU value is always FREE. So I hardcoded the sku variable value to 'FREE' in the below script

/usr/local/Cellar/azure-cli/2.0.76/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/utils.py

def _normalize_sku(sku):
    **sku = 'FREE'**
    sku = sku.upper()
    if sku == 'FREE':
        return 'F1'
    if sku == 'SHARED':
        return 'D1'
    return sku

I then ran this command to successfully deploy my app
az webapp up --sku FREE -n <myappname>-l centralus

Please make sure that the above function can gracefully handle  null values too. Customers sometimes forget to  add sku value in the command line.

**Comment by angel-hernandez91** [comment]
> this is fixed by #11019

This is still broken if you're running things from the cli that available as a webapp on portal.azure.com
